README.md

```diff
@@ -179,88 +299,378 @@ It will come with those slots, very related to what we have in the Nunjucks bloc
+> They are efficiently compiled, support TypeScript and are only executed
+> once even if the component appears may times on the page. Defining a
+> web component allows us to leverage that code more than once by using
+> the newly defined element as often as we wish.
+
+## `aposSetQueryParameter`: working with query parameters
+
+One last thing: query parameters. Sometimes we want to create pagination
+links with page numbers, add filters to a URL's query string, and so on.
+But, working with query parameters coming from Apostrophe can
+be a little bit tricky because there are often special query parameters
+present during editing that should not be part of a visible URL.
+
+As a convenience, Apostrophe provides `aposSetQueryParameter` to abstract
+all that away.
+
+Here is how the `BlogIndexPage.astro` component of the [apostrophecms/astro-frontend]
+(https://github.com/apostrophecms/astro-frontend) project generates
+links to the each page of blog posts:
+
+```js
+---
+import setParameter from '@apostrophecms/apostrophe-astro/lib/AposSetQueryParameter.js';
+
+const {
+  pieces,
+  currentPage,
+  totalPages
+} = Astro.props.aposData;
+
+const pages = [];
+for (let i = 1; (i <= totalPages); i++) {
+  pages.push({
+    number: i,
+    current: page === currentPage,
+    url: setParameter(Astro.url, 'page', i)
+  });
 }
+---
+
+<section class="bp-content">
+  <h1>{ page.title }</h1>
+
+  <h2>Blog Posts</h2>
+
+  {pieces.map(piece => (
+    <h3>
+      <a href={ piece._url }>{ piece.title }</a>
+    </h3>
+  ))}
+
+  {pages.map(page => (
+    <a
+      class={(page === currentPage) ? 'current' : ''} 
+      href={page.url}>{page.number}
+    </a>
+  ))}
+</section>
+```
+
+Imported here as `setParameter`, `aposSetQueryParameter` allows
+us to do two things:
+
+1. Take a URL and return a new URL with a certain query parameter set
+to a new value.
+2. Remove a query parameter completely by passing the empty string as
+a value, or by passing `null` or `undefined`.
+
+While you can get the same result by manipulating `Astro.url` yourself,
+you'll be able to avoid the confusing presence of query parameters
+like `aposMode` by using this convenient feature.
+
+## What about Vue, React, SvelteJS, etc.?
+
+While not shown directly in the examples above, **Astro can import components
+written in any of these frameworks.** Just use `astro add` to install
+the appropriate integration, then `import` your components freely in your
+`.astro` files. For complete documentation and examples, see the
+[`@astrojs/react` integration](https://docs.astro.build/en/guides/integrations-guide/react/).
+
+In this way, Astro acts as a **univeral bridge** to essentially all modern
+frontend frameworks.
+
+## A note on production use
+
+For production use, any Astro hosting adapter that supports `mode: 'server'` should
+be acceptable. In particular, our [apostrophecms/astro-frontend]
+(https://github.com/apostrophecms/astro-frontend) project comes pre-configured
```

Comment:
This link is malformed. The space between the brackets and parentheses needs to be removed.